tinyyaml.lua

```diff
@@ -337,7 +337,12 @@ function Parser:parseflowstyle(line, lines)
       if not s then
         error('invalid flowstyle line: '..line)
       end
-      tinsert(stack, {v=s, t='s'})
+      local n = tonumber(s)
+      if n and not startswith(line, '"') then
```

Comment:
1. why add the condition `not startswith(line, '"')`
2. why not add the condition `not startswith(line, ''')`

Reply:
1. because number  type should  ` not startwith '"' `
2. shoud add   the condition `not startswith(line, ''')`

Reply:
why not use `c == '"'`?

Reply:
yes,  use `c == '"'`  more suitably

Reply:
I think we'd better fix this in `Parser:parsestring` function, the `parsestring` should return the correct value

Reply:
We can add this in the end of `parsestring`
```
  buf = rtrim(buf)
  local val = tonumber(buf) or buf
  return val, line
```

Reply:
Thanks for the review! 
Yes this will be better as will also convert numbers like `4.3e+3` and not only numbers inside `{}` or `[]`